aquamarine/src/plumber.rs

```diff
@@ -168,6 +171,16 @@ impl<RT: AquaRuntime, F: ParticleFunctionStatic> Plumber<RT, F> {
                 particle_id = particle.particle.id,
             ),
         }
+        self.wake();
+    }
+
+    pub fn on_worker_created(&mut self, worker_id: WorkerId, thread_count: usize) {
+        let vm_pool = VmPool::new(thread_count, self.config.clone(), None, None); // TODO: add metrics
+        self.worker_vm_pools.insert(worker_id, vm_pool);
+    }
+
+    pub fn on_worker_removed(&mut self, worker_id: WorkerId) {
```

Comment:
Let's avoid `on_event_x` naming, it expresses API of one entity in terms of other's entity internals. Very hard to understand.

```suggestion
    pub fn remove_worker_pool(&mut self, worker_id: WorkerId) {
```